### DISCUSSION TITLE
bekomme es nicht zum laufen evcc und solectrus

### DISCUSSION THREAD

**Discussion by ZAG-SH** [solectrus/solectrus#3656] [General]
Hi,
ich bin den ganzen Tag am Lesen und Videos schauen.
Kurz-, Intensivstudium für einen nicht-ITler ;-)

Mit  Freude habe ich SOLECTRUS gefunden.

EVCC ist installiert und läuft MQTT über HIVEMQ angelegt und einen Docker installiert.

Bei dem Befehl "docker compose up -d" ist jedes mal Schluss und meine PI friert ein (der Prozeß der Installation wird abgebrochen, Netwerk ist weg und danach startet die PI nicht mehr / rund. (entweder kommen jede Menge Fehler, oder es bleibt beim Mauszeiger und schwarzem Bildschirm).

Was mache ich da falsch ??

Und noch eine Frage zu dem Konfigurator: was gebe ich bei den Topic an (welcher Pfad genau, oder ist es egal) ??

Danke !

**Comment by ledermann** [comment]
Wenn der Raspi unerwartet neu bootet, ist das natürlich ein schlechtes Zeichen. Es könnte auf Speichermangel hinweisen. Oder Hardware-Probleme.

Wichtig wäre, die genauen Fehlermeldungen zu kennen. Versuche mal, statt dem `docker compose up -d` ein `docker compose up` zu machen (also ohne `-d`), dann läuft alles im Vordergrund und die Logs werden ausgegeben. Bitte poste den exakten Output, der da kommt - falls das überhaupt geht und nicht wieder ein Reboot erfolgt.

Außerdem mal mit `free -h` den freien Speicherplatz ermitteln. Auch hier das Ergebnis bitte posten.

Bezüglich Konfigurator: Was bei den Topics einzugeben ist, ist keineswegs egal. Es ist anzugeben, wo genau beim MQTT-Broker der jeweilige Messwert zu finden ist. Bei evcc sind das Angaben wie `evcc/site/homePower` usw.

**Comment by ZAG-SH** [comment]
Hier schon mal der Speicher:

```
              gesamt     benutzt        frei     gemns.   Puffer/Cache   verfügbar
Speicher:      907Mi       408Mi        97Mi     8,4Mi           462Mi       498Mi
Swap:          199Mi        10Mi       189Mi
```

Und die Meldung, wenn ich compose eingebe:

```
pi@raspberrypi:~ $ docker compose up
yaml: line 9: could not find expected ':'

pi@raspberrypi:~ $ docker compose up -d
yaml: line 9: could not find expected ':'
```

**Reply by ledermann** [reply]
1 GB RAM sind arg wenig. Ich habe die untere Grenze an freiem Arbeitsspeicher bislang nicht ermittelt, aber ein Raspi mit mindestens 2 GB sollte es schon sein. Hier mal der Speicherverbrauch von SOLECTRUS auf meinem Raspi 4B mit 8 GB RAM:

```
$ docker compose stats --no-stream
CONTAINER ID   NAME                          CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O         PIDS
5edc2bb69442   solectrus-dashboard-1         0.03%     166.9MiB / 7.627GiB   2.14%     305kB / 872kB     37.8MB / 32.8kB   12
c538c7bc2662   solectrus-power-splitter-1    0.00%     45.83MiB / 7.627GiB   0.59%     37.6kB / 45.9kB   13.1MB / 0B       4
f328fa7a773c   solectrus-senec-collector-1   0.00%     38.11MiB / 7.627GiB   0.49%     99.6kB / 39.2kB   15.7MB / 0B       5
54f3f559241a   solectrus-redis-1             3.10%     16.51MiB / 7.627GiB   0.21%     148kB / 81.4kB    13.4MB / 41kB     7
c0392a5c0ca5   solectrus-influxdb-1          0.10%     185.5MiB / 7.627GiB   2.38%     70.2kB / 59kB     112MB / 246kB     19
ece080c81d04   solectrus-postgresql-1        0.00%     45.18MiB / 7.627GiB   0.58%     126kB / 108kB     30.5MB / 762kB    6
dddf83831d20   solectrus-watchtower-1        0.00%     15.79MiB / 7.627GiB   0.20%     776B / 0B         12.7MB / 0B       9
```

Zählt man die Angaben unter `MEM USAGE` zusammen, kommt man auf ca. 512 MB. Bei dir sind hingegen nur 498 MB frei. Das ist alles sehr, sehr eng.

Die Sache mit dem Syntax-Fehler in der `compose.yaml` ist ein ganz anderes Problem. Bist du sicher, dass du die `compose.yaml` korrekt aus dem Konfigurator übernommen hast? Das YAML-Format ist empfindlich gegenüber Formatierungsfehlern. Falls du den Fehler nicht selbst findest, poste mal die `compose.yaml` hier.

**Reply by ZAG-SH** [reply]
Hallo,
danke noch einmal für Deine schnelle Antwort.
Heute ist meine neue Pi (mit 4GB) angekommen !

Ich habe jetzt alles noch einmal von vorne begonnen...

Noch ein/zwei ;-)  Fragen:
-  zu den MQTT Topics: wo finde ich die Angaben genau ? Auf der evcc Seite sind leider nicht viele Angaben bei evcc/site/...

Ich habe jetzt sogar einen ersten Zugriff über die IP bekommen :-) , aber sind ja noch vieele Fehler beim Installieren gewesen
![solectrus](https://github.com/user-attachments/assets/73fc5926-d4d1-4fb7-80c7-b68927eddd22)

Danke !

**Reply by ledermann** [reply]
Ok, mit 4GB RAM sollte es keine Probleme geben. 

Die Anbindung an evcc erfordert einen MQTT-Broker, d.h. evcc schickt die Daten an den Broker, wo SOLECTRUS sich das dann abholt.

Genaueres findest du hier:
https://docs.evcc.io/docs/integrations/home-assistant
https://docs.evcc.io/docs/integrations/mqtt-api

**Reply by ZAG-SH** [reply]
Danke !  

Ich habe noch eine Verständnisfrage zu den topics im Konfigruator:

Unter Umstellung habe ich auf der Seite jetzt die Bezeichnungen der Sensoren gefunden.

Im Konfigruator ist jedoch für Einspeisung und Bezug, oder Batterie Ent, -beladung nur ein Eintrag möglich. Bei den Sensoren haben alle einen eigenen Namen.

Danke !

**Reply by ZAG-SH** [reply]
Evtl ist es gut zu wissen, dass ich eher Fraktion "Bürofutzi" bin und kein IT ler.

Wenn es zu schlimm wird, bitte einfach Bescheid sagen.

Und wenn ich es hinbekommen sollte, werde ich eine bebilderte Anleitung "evcc & solectrus auf der raspi für dummies" erstellen ;-) .

**Reply by ledermann** [reply]
Ok, möglicherweise deckt der Konfigurator nicht jede erdenkliche Möglichkeit ab. Durch manuelle Einstellungen kann der MQTT-Collector weit mehr. 

Ich schlage vor, dass du schrittweise vorgehst, d.h. zunächst mit dem Konfigurator eine Grundkonfiguration erstellst (unklare Topics leer lassen) und im Dashboard überprüfst. Dann den Konfigurator ganz beiseite lassen und die fehlenden Sensoren manuell ergänzen, aber immer nur einen nach dem anderen.

Zur Konfiguration des MQTT-Collectors gibt es hier weitere Informationen:
https://docs.solectrus.de/referenz/mqtt-collector/konfiguration

Wenn du überhaupt nicht klarkommst, dann poste mal deine Grundkonfiguration (`.env` und `compose.yaml`) eine Liste der Topics, die du anbinden möchtest und ich passe dann deine Konfiguration an.

**Reply by ZAG-SH** [reply]
Vielen Dank !

Ich werde am Wochenende mal rumprobieren und mich weiter "reinlesen"...muss doch gehen. 

Sollte ich völlig verzweifeln, komme ich gerne auf Dein Angebot zurück.

**Comment by ZAG-SH** [comment]
Hi,
bin am Basteln ;-)

Auf dem Dashboard steht immer, dass ich das Passwort eingeben soll (Hinweis: ist ADMIN_PASSWORD in der ,env).

Das nehme ich und es kommt immer: "nicht gültig" (ich habe es schon mal auf 1234 geändert, aber auch das ohne Erfolg)

**Reply by ledermann** [reply]
Vermutlich steht das Passwort noch auf dem Default "my-secret-login-password". 

Nach einer Änderung der `.env` (oder der `compose.yaml`) müssen die Container zwingend neu erstellt werden, z.B. mit `docker compose up -d`. Falls das nicht helfen sollte, mach vorher noch ein `docker compose down`.

**Comment by ZAG-SH** [comment]
Guten Morgen,
mir raucht der Kopf, aber es wird immer besser mit dem Verstehen (teilweise ;-) ).

Bisher habe ich immer "keine Verbindung" gehabt, durch den Hinweis auf die MQTT Eingaben, habe ich den Configruator jetzt ein paarmal gefüttert.
Mal mit einem Topic, mal mehrere. Das Ergebnis ist immer gleich.

Docker compose up läuft in einer Dauerschleife.

Dabei sind mir 2 Sachen aufgefallen:

 power-splitter : bundler: failed to load command: app.rb 

Und ich wundere mich über diese IP (meine Pi hat 177) in daghboard 1:

 I, [2024-11-24T12:40:20.373808 #1]  INFO -- : [192.168.178.139] method=GET path=/charts/inverter_power/now format=html controller=ChartsController action=index status=200 allocations=24337 duration=439.26 view=434.45 

Irgendetwas habe ich wohl falsch eingegeben (?)
Danke

**Reply by ledermann** [reply]
Der Fehler beim Power-Splitter ist nicht vollständig wiedergegeben, da müsste mehr Text stehen.

Die IP `...139` vs. `...177` liegt vermutlich an den zwei parallel laufenden Containern. Einer ist zu viel, siehe mein anderer Kommentar.

**Comment by ZAG-SH** [comment]
![image](https://github.com/user-attachments/assets/e6a69e17-4fdd-410c-bb4d-fdeabc7fdf07)

Hier die Container --- ist alles auf grün.

**Reply by ledermann** [reply]
Hier ist ein überzähliger Container zu sehen: `pi-app-1` und `solectrus-dashboard-1` beziehen sich beide auf das gleiche Image. Einer muss weg. Überhaupt ist da etwas Durcheinander: Einige Container beginnen mit `pi-`, andere hingegen mit `solectrus-`.  Das sind vermutlich Reste von verschiedenen Experimenten. Besser mal alle Container herunterfahren und dann neu mit `docker compose up -d` starten.

Auch weiß ich nicht, ob du dir mit der Verwendung von Portainer einen Gefallen tust. Das irritiert möglicherweise mehr, als es hilft. Besser ist, rein auf der Konsole zu arbeiten.

**Reply by ZAG-SH** [reply]
hej, danke...werde ich gleich mal ausprobieren.

**Comment by ZAG-SH** [comment]
Hat leider nicht geklappt, daher habe ich alles neu installiert.
Auch wieder mit Fehlern, aber jetzt:
![image](https://github.com/user-attachments/assets/84c053ad-8629-4379-a64f-4aac1a4428cb)

Die erste Anzeige :-) (JUHUU!)

Nun muss ich mich langsam an alle Sensoren annähern...

**Comment by ZAG-SH** [comment]
Hi ,

mal bekomme ich einen Serverfehler und mal nicht.

Heute morgen wurde mir der Hausverbrauch angezwigt (mehr ist bei mir noch nicht zu sehen), heute Nachmittag dann server-error.

In den logs habe ich folgendes gefunden:

dashboard-1  | [192.168.178.139] ActionView::Template::Error (percent must be be                                                                                                                                                                                                                                             tween 0 and 100, got -240.35):
dashboard-1  | [192.168.178.139]
dashboard-1  | Causes:
dashboard-1  | [192.168.178.139] ArgumentError (percent must be between 0 and 10                                                                                                                                                                                                                                             0, got -240.35)

Das bedeutet wohl , dass falsche %Zahlen angegeben wurden...aber wo ?? (werde .ennv, noch compose.yaml...

Danke !

**Reply by ledermann** [reply]
Hm, das liegt wohl daran, dass du momentan nur den Hausverbrauch und den Akku-Ladestand hast. SOLECTRUS führt etliche Berechnungen durch, z.B. Autarkie und Eigenverbrauch. Offenbar gibt es in deiner Konfiguration eine Situation, bei der ein ungültiger Wert berechnet wird (-240 statt irgendwas zwischen 0 und 100). Ich habe versucht, das zu reproduzieren, das gelingt mir aber nicht. Hilfreich wäre es, wenn du deine `.env` posten könntest - oder den Output von `docker compose config`.

Ansonsten schlage ich vor, dass du die fehlenden Sensoren ergänzt, insbesondere `INVERTER_POWER` - dann wird dieses Problem sich vermutlich von allein lösen.

Unabhängig davon werde ich aber versuchen, SOLECTRUS robuster zu machen, sodass fehlende Sensoren keinen Fehler 500 verursachen.

**Reply by ZAG-SH** [reply]
Hallo,

danke für die schnelle Rückmeldung !

Da werde ich nachher mal basteln.

**Reply by ledermann** [reply]
Noch eine Ergänzung: Möglicherweise hast du einen Vorzeichenfehler in einem Sensor. Du kannst dir genau ansehen, welche Werte der MQTT-Collector empfängt und schreibt, und zwar wie folgt:

```
docker compose logs mqtt-collector
```

Da wird eine Menge Zeug protokolliert, das zum Debugging hilfreich sein könnte.

Ansonsten habe ich gerade noch eine Änderung gemacht, dass die Autarkie und der Eigenverbrauch korrigiert werden, wenn der Wert nicht zwischen 0% - 100% liegt. Wenn du magst, kannst du auf die Entwickler-Version wechseln (`image: ghcr.io/solectrus/solectrus:develop` statt `:latest`). Aber wie gesagt: Die eigentliche Ursache für den Fehler 500 liegt in deinen Sensoren.

**Reply by ZAG-SH** [reply]
danke, ja ,  glaube ich gerne. Denn jetzt habe ich schon mehr Anzeigen, aber die 2,x x sind Hausbverbrauch, und die 5 W Ladung der Batterie (habe ich aus der openWB Software, die derzeit noch läuft)

![image](https://github.com/user-attachments/assets/e9d7f897-6b41-4c4f-8460-8972aa8c9902)

**Reply by ledermann** [reply]
Ok, somit hast du die Sensoren vertauscht. Hilfestellung ist schwierig, wenn ich die Topics nicht kenne, die es bei in deinem Setup gibt. Grundsätzlich ist es so:

1) Der MQTT-Collector empfängt Topics und schreibt sie nach InfluxDB unter Angabe von Measurement und Field. Definiert wird das in der `.env` über Variablen, deren Name mit `MAPPING_` beginnt.
2) Für das Dashboard legst du dann mittels Sensoren fest, wo in InfluxDB (wieder über Measurement und Field) ein Wert zu finden ist.

Offenbar wurde bei dir nun etwas vertauscht. Da überhaupt Werte angezeigt werden, ist die Entgegennahme per MQTT erfolgreich gewesen und es sind nur die Sensoren für das Dashboard zu korrigieren.

> die 2,x x sind Hausbverbrauch, und die 5 W Ladung der Batterie 

Das kannst du so korrigieren:

```
INFLUX_SENSOR_HOUSE_POWER=Measurement:Field, das derzeitig bei INFLUX_SENSOR_INVERTER_POWER steht
INFLUX_SENSOR_BATTERY_CHARGING_POWER=Measurement:Field, das derzeitig bei INFLUX_SENSOR_HOUSE_POWER steht
```

Hier ein Link zur vollständigen Liste der verfügbaren Sensoren:
https://docs.solectrus.de/referenz/dashboard/konfiguration#umgebungsvariablen-f%C3%BCr-die-sensor-konfiguration-ab-version-015

**Comment by ZAG-SH** [comment]
Guten Morgen,

leider hatte ich nach der Anpassung nur noch einen Wert. Daher habe ich den Konriguator neu ausgeführt.

Ich habe auch auf develop umgestellt ;-)

Wenn ich auf Einzelwerte gehe, wird jetzt auch die Erzeugung richtig angezeigt. Der Server-Fehler ist leider noch da und vermutlich fehlen die Angaben zur Autarkie u.ä.

Oder auch die Bezeichnung der topics (??)

Bei github steht es so: 

z.B. MQTT_TOPIC_HOUSE_POW=evcc/site/homePower

meine (vom Konfigruator erzeugt): MAPPING_1_TOPIC=evcc/site/homePower

(obwohl er bei mir ja die Erzeugung jetzt richtig anzeigt)...?

Hier die Einträge von meiner neuen .env
[env (1).txt](https://github.com/user-attachments/files/17964716/env.1.txt)

Danke

**Reply by ledermann** [reply]
In deiner `.env` sind mir zwei Dinge aufgefallen:

1. Beim Mapping von `evcc/site/gridPower` sind die Vorzeichen vertauscht. Laut Doku von evcc steht ein negativer Wert für Einspeisung und ein positiver Wert für Netzbezug:

    > [...] evcc benutzt positive (+) Werte für eingehende Ströme (Netzbezug, PV Erzeugung, Hausbatterie-Entladung) und negative (-) Werte für ausgehende Ströme (Netzeinspeisung, PV Wechselrichter Ruhestrombedarf oder Hausbatterie-Ladung) [...]
    > https://docs.evcc.io/docs/reference/configuration/meters

2. Es fehlt ein Mapping für den Hausakku bzw. du hast Ladung/Entladung mit Ladestand verwechselt.

Ich habe deine `.env` überarbeitet und korrigiert. Hier die **relevanten** Ausschnitte mit Kommentierung:

```env
##################################################################
###                       MQTT Collector                       ###
##################################################################

# PV-Erzeugung
MAPPING_0_TOPIC=evcc/site/pvPower
MAPPING_0_MEASUREMENT=pv
MAPPING_0_FIELD=inverter_power
MAPPING_0_TYPE=integer

# Hausverbrauch
MAPPING_1_TOPIC=evcc/site/homePower
MAPPING_1_MEASUREMENT=pv
MAPPING_1_FIELD=house_power
MAPPING_1_TYPE=integer

# Netzbezug und Einspeisung
MAPPING_2_TOPIC=evcc/site/gridPower
MAPPING_2_MEASUREMENT_POSITIVE=pv
MAPPING_2_MEASUREMENT_NEGATIVE=pv
MAPPING_2_FIELD_POSITIVE=grid_import_power
MAPPING_2_FIELD_NEGATIVE=grid_export_power
MAPPING_2_TYPE=integer

# Ladestand Hausakku
MAPPING_3_TOPIC=evcc/site/batterySoc
MAPPING_3_MEASUREMENT=pv
MAPPING_3_FIELD=battery_soc
MAPPING_3_TYPE=float

# Wallbox-Leistung
MAPPING_4_TOPIC=evcc/site/loadpoints/1/chargePower
MAPPING_4_MEASUREMENT=pv
MAPPING_4_FIELD=wallbox_power
MAPPING_4_TYPE=integer

# Akku-Ladung und -Entladung
MAPPING_5_TOPIC=evcc/site/batteryPower
MAPPING_5_MEASUREMENT_POSITIVE=pv
MAPPING_5_MEASUREMENT_NEGATIVE=pv
MAPPING_5_FIELD_POSITIVE=battery_discharging_power
MAPPING_5_FIELD_NEGATIVE=battery_charging_power
MAPPING_5_TYPE=integer

##################################################################
###                      Sensor mapping                        ###
##################################################################

# Verwendete Sensoren 
INFLUX_SENSOR_INVERTER_POWER=pv:inverter_power
INFLUX_SENSOR_HOUSE_POWER=pv:house_power
INFLUX_SENSOR_GRID_IMPORT_POWER=pv:grid_import_power
INFLUX_SENSOR_GRID_EXPORT_POWER=pv:grid_export_power
INFLUX_SENSOR_BATTERY_SOC=pv:battery_soc
INFLUX_SENSOR_WALLBOX_POWER=pv:wallbox_power
INFLUX_SENSOR_BATTERY_CHARGING_POWER=pv:battery_charging_power
INFLUX_SENSOR_BATTERY_DISCHARGING_POWER=pv:battery_discharging_power

# Nicht verwendete Sensoren
INFLUX_SENSOR_WALLBOX_CAR_CONNECTED=
INFLUX_SENSOR_CASE_TEMP=
INFLUX_SENSOR_SYSTEM_STATUS=
INFLUX_SENSOR_SYSTEM_STATUS_OK=
INFLUX_SENSOR_GRID_EXPORT_LIMIT=
INFLUX_SENSOR_INVERTER_POWER_FORECAST=
INFLUX_SENSOR_HEATPUMP_POWER=
INFLUX_SENSOR_CAR_BATTERY_SOC=

# Aus dem Hausverbrauch herauszurechnende Sensoren
INFLUX_EXCLUDE_FROM_HOUSE_POWER=WALLBOX_POWER
```

Es sind also Variablen für ein **sechstes** Mapping hinzugekommen. In deiner `compose.yaml` musst du diese **ergänzen** (unter `services => mqtt-collector => environment`), und zwar unter die bestehenden Variablen für Mapping 0 bis 4:

```yaml
- MAPPING_5_TOPIC
- MAPPING_5_MEASUREMENT_POSITIVE
- MAPPING_5_MEASUREMENT_NEGATIVE
- MAPPING_5_FIELD_POSITIVE
- MAPPING_5_FIELD_NEGATIVE
- MAPPING_5_TYPE
```

Nach der Änderung in `.env` und `compose.yaml` musst du alles neu starten (`docker compose up -d`). Am besten löschst du anschließend auch die Tageszusammenfassungen (im UI als Admin einloggen, dann unter Einstellungen den Button "Zurücksetzen" klicken), weil sonst in vergangenen Zeiträumen weiterhin falsche Angaben erscheinen können.

Wenn du an irgendeiner Stelle einen Fehler 500 bekommst, brauche ich immer zwingend einen Auszug aus dem Log und am besten auch die URL, die du aufgerufen hast, sonst kann ich die Ursache nicht finden. Und auch die exakte Versionsnummer.

Falls weiterhin nicht korrekte oder nicht alle erwarteten Werte angezeigt werden, schaue unbedingt in das Log des MQTT-Collectors, am besten so:

```bash
docker compose logs mqtt-collector -f
```

Ich hoffe, das hilft dir nun weiter.

**Reply by ZAG-SH** [reply]
Guten Morgen,

ich habe die Änderungen, die Du mir genannt hast, eingepflegt und: ES LÄUFT !!

vielen , vielen Dank !!!

(ich werde mich die Tage noch ein bisschen damit beschäftigen, damit ich es einigermaßen auch nachvollziehen kann ;-) )

_**Ich lade gerade das Auto, daher ist ein so hoher Bezug ;-)**_(das muss ich wohl noch anpassen, damit das Auto auch erscheint)



![image](https://github.com/user-attachments/assets/340c7de5-12eb-4926-af6c-40196e2ac1c9)

**Reply by ledermann** [reply]
Ok, schonmal sehr gut. Wegen der fehlenden Wallbox-Beladung müsstest du nochmal das MAPPING_4 prüfen. Erster Anlaufstelle ist das Log des MQTT-Collectors. Denkbar wäre, dass es das Topic `evcc/site/loadpoints/1/chargePower` so nicht gibt. Vielleicht müsste es `0` statt `1` heißen, oder sonst wie anders. Das müsstest du bei evcc abklären.

**Reply by ZAG-SH** [reply]
Danke für Deine Unterstützung !

Ich habe es bei evcc ins Forum gestellt. Falls ich eine Antwort bekomme, werde ich es testen und hier dann auch posten.

**Reply by frizzle0815** [reply]
Das einfachste wäre wohl einmal z.B. MQTT Explorer (über Google Suche zu finden) zu installieren und zu schauen, welche Topics Evcc zur Verfügung stellt. Dann müsstest Du eigentlich recht schnell finden, was Du brauchst.

**Comment by ZAG-SH** [comment]
Tatsächlich musste ich /site/ herausnehmen.

Mit MAPPING_4_TOPIC=evcc/loadpoints/1/chargePower wird jetzt die Ladung des E-Autos angezeigt ! Danke !

Aber der Hausverbrauch war dann weg :-) 

Die Lösung dazu: den Hinweis exclude ausschalten: #INFLUX_EXCLUDE_FROM_HOUSE_POWER=WALLBOX_POWER

![image](https://github.com/user-attachments/assets/220947ca-4967-42ab-a10d-8b7ff0fe06bb)

(soll ich die Bilder lieber löschen ?)
